src/app/pages/sharing/smb/smb-form/smb-form.component.ts

```diff
@@ -1,652 +1,411 @@
-          if (!res) {
-            this.stripACLWarningSent = true;
-            this.showStripAclWarning();
-          }
-        });
-      }
-    });
-
-    setTimeout(() => {
-      if (entityForm.formGroup.controls['timemachine'].value) {
-        this.isTimeMachineOn = true;
-      }
-    }, 700);
-
-    entityForm.formGroup.controls['purpose'].valueChanges.pipe(untilDestroyed(this)).subscribe((res) => {
-      this.clearPresets();
-      for (const item in this.presets[res].params) {
-        this.presetFields.push(item as keyof SmbShare);
-        const ctrl = entityForm.formGroup.controls[item];
-        if (ctrl && item !== 'auxsmbconf') {
-          ctrl.setValue(this.presets[res].params[item as keyof SmbShare]);
-          ctrl.disable();
-        }
-      }
-    });
+    this.presetFields = [];
   }
 
-  afpConfirm(): void {
-    const afpControl = this.entityForm.formGroup.controls['afp'];
+  setSmbShareForEdit(smbShare: SmbShare): void {
+    this.existingSmbShare = smbShare;
+    this.title = helptextSharingSmb.formTitleEdit;
+    const index = this.namesInUse.findIndex((name) => name === smbShare.name);
+    if (index >= 0) {
+      this.namesInUse.splice(index, 1);
+    }
+    this.form.patchValue(smbShare);
+  }
 
+  afpConfirmEnable(value: boolean): void {
+    if (!value) {
+      return;
+    }
+    const afpControl = this.form.get('afp');
     this.dialog.confirm({
       title: helptextSharingSmb.afpDialog_title,
       message: helptextSharingSmb.afpDialog_message,
       hideCheckBox: false,
       buttonMsg: helptextSharingSmb.afpDialog_button,
       hideCancel: false,
     }).pipe(untilDestroyed(this)).subscribe((dialogResult: boolean) => {
-      if (dialogResult) {
-        afpControl.setValue(!afpControl.value);
+      if (!dialogResult) {
+        afpControl.setValue(!value);
       }
     });
   }
 
-  showStripAclWarning(): void {
-    this.dialog.confirm({
-      title: helptextSharingSmb.stripACLDialog.title,
-      message: helptextSharingSmb.stripACLDialog.message,
-      hideCheckBox: true,
-      buttonMsg: helptextSharingSmb.stripACLDialog.button,
-      hideCancel: true,
+  submit(): void {
+    this.isLoading = true;
+    this.cdr.markForCheck();
+    const smbShare = this.form.value;
+    let request$: Observable<unknown>;
+    if (this.isNew) {
+      request$ = this.ws.call('sharing.smb.create', [smbShare]);
+    } else {
+      request$ = this.ws.call('sharing.smb.update', [this.existingSmbShare.id, smbShare]);
+    }
+
+    request$.pipe(
+      untilDestroyed(this),
+    ).subscribe(() => {
+      this.shouldServiceRestart().pipe(
+        switchMap((restart) => {
+          if (restart) {
+            return this.restartServices();
+          }
+          return of(true);
+        }),
+        switchMap(this.shouldRedirectToAclEdit),
+        filter(Boolean),
+        switchMap(this.shouldEnableServiceAutomaticRestart),
+        switchMap(this.shouldStartAndUpdateCifsService),
```

Comment:
I already messaged someone from middleware for that reason. Should be able to get a response tomorrow.

Reply:
Couldn't get a response from middleware. I've made changes to address the issues. Not sure how exactly locked dataset comes into play when enabling or disabling the service. Gonna wait for a response from middleware.

Reply:
Quoting from @anodos325 

> Okay, the reasoning behind a service restart after applying change is whether the change should impact existing users. In an enterprise / business environment, making a service restart can be very undesirable and so we want to avoid it whenever possible.

> The reasons why we want to restart is when something has change requiring us to force everyone to restart:

- new time machine share - we need to change how the SMB server advertises itself (and clients need to see this change)

- hosts allow / hosts deny - security settings of share have changed and we need restart in order to enforce them

- home share - this changes details of how paths / shares are evaluated. Clients need to refresh this

- any path change - clients won't see this parameter change until they're forced to negotiate a new session

> The user should be prompted for a service restart so that they have a choice about whether to do it.... think of environment with 1000 users actively on server. We don't want to boot them unless administrator says "go for it".